<a href="https://colab.research.google.com/github/MaximTislenko/DS_in_MED/blob/main/Less_02_06.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import os
import pandas as pd
import numpy as np

In [2]:
!wget https://github.com/MaximTislenko/DS_in_MED/raw/main/clinical.project-tcga-dlbc.2023-09-20.tar.gz

--2023-10-04 19:36:44--  https://github.com/MaximTislenko/DS_in_MED/raw/main/clinical.project-tcga-dlbc.2023-09-20.tar.gz
Resolving github.com (github.com)... 140.82.112.3
Connecting to github.com (github.com)|140.82.112.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/MaximTislenko/DS_in_MED/main/clinical.project-tcga-dlbc.2023-09-20.tar.gz [following]
--2023-10-04 19:36:45--  https://raw.githubusercontent.com/MaximTislenko/DS_in_MED/main/clinical.project-tcga-dlbc.2023-09-20.tar.gz
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 9344 (9.1K) [application/octet-stream]
Saving to: ‘clinical.project-tcga-dlbc.2023-09-20.tar.gz’

clinical.project-tc 100%[===================>]   9.12K  --.-KB/s   

In [3]:
import tarfile
zipfile = '/content/clinical.project-tcga-dlbc.2023-09-20.tar.gz'
if zipfile.endswith("tar.gz"):
    tar = tarfile.open(zipfile, "r:gz")
elif zipfile.endswith("tar"):
    tar = tarfile.open(zipfile, "r:")
tar.extractall()
tar.close()

##Видимо клиническая картина
Столбцы:
* 'case_id' - **идентификатор амбулаторной/стационарной карты**,
* 'case_submitter_id'- **идентификатор источника данных**,
* 'project_id' - **идентификатор проекта**,
* 'age_at_index' - **возраст на момент описания клиники**,
* 'age_is_obfuscated' - **возраст неизвестен?**,
* 'cause_of_death' - **причина смерти**,
* 'cause_of_death_source' - **причина смерти источник**,
* 'country_of_residence_at_enrollment' - **страна проживания на момент регистрации**,
* 'days_to_birth' - **дата рождения**,
* 'days_to_death' - **дата смерти**,
* 'ethnicity' - **этническая принадлежность**,
* 'gender' - **пол**,
* 'occupation_duration_years' - **трудовой стаж, лет**,
* 'premature_at_birth' - **недоношенный при рождении**,
* 'race' - **расса**,
* 'vital_status' - **жизненый уровень**,
* 'weeks_gestation_at_birth' - **неделя гестации при рождении**,
* 'year_of_birth' - **год рождения**,
* 'year_of_death' - **год смерти**,
* 'adrenal_hormone' - **гормоны надпочечников** ,
* 'age_at_diagnosis' - **продолжительность болезни**,

#AJCC - Американский объединенный комитет по раку
* 'ajcc_clinical_m'- **продолжительность проявления клин.проявлений, мес??**,
* 'ajcc_clinical_n'- **продолжительность проявления клин.проявлений, нед??**,
* 'ajcc_clinical_stage' - **продолжительность стадии клин.проявлений??**
* 'ajcc_clinical_t' - **продолжительность проявления клин.проявлений, час??**,
* 'ajcc_pathologic_m'- **продолжительность заболевания, мес**,
* 'ajcc_pathologic_n'- **продолжительность заболевания, нед**,
* 'ajcc_pathologic_stage - **продолжительность стадии заболевания**',
* 'ajcc_pathologic_t'- **продолжительность заболевания, час**,
* 'ajcc_staging_system_edition - ',
* 'anaplasia_present',
* 'anaplasia_present_type',
* 'ann_arbor_b_symptoms',
* 'ann_arbor_b_symptoms_described',
* 'ann_arbor_clinical_stage',
* 'ann_arbor_extranodal_involvement',
* 'ann_arbor_pathologic_stage',
* 'best_overall_response',
* 'breslow_thickness',
* 'burkitt_lymphoma_clinical_variant',
* 'child_pugh_classification',
* 'circumferential_resection_margin',
* 'classification_of_tumor',
* 'cog_liver_stage',
* 'cog_neuroblastoma_risk_group',
* 'cog_renal_stage',
* 'cog_rhabdomyosarcoma_risk_group',
* 'days_to_best_overall_response',
* 'days_to_diagnosis',
* 'days_to_last_follow_up',
* 'days_to_last_known_disease_status',
* 'days_to_recurrence',
* 'eln_risk_classification',
* 'enneking_msts_grade',
* 'enneking_msts_metastasis',
* 'enneking_msts_stage',
* 'enneking_msts_tumor_site',
* 'esophageal_columnar_dysplasia_degree',
* 'esophageal_columnar_metaplasia_present',
* 'figo_stage',
* 'figo_staging_edition_year',
* 'first_symptom_prior_to_diagnosis',
* 'gastric_esophageal_junction_involvement',
* 'gleason_grade_group',
* 'gleason_grade_tertiary',
* 'gleason_patterns_percent',
* 'goblet_cells_columnar_mucosa_present',
* 'greatest_tumor_dimension',
* 'gross_tumor_weight',
* 'icd_10_code',
* 'igcccg_stage',
* 'inpc_grade',
* 'inpc_histologic_group',
* 'inrg_stage',
* 'inss_stage',
* 'international_prognostic_index',
* 'irs_group',
* 'irs_stage',
* 'ishak_fibrosis_score',
* 'iss_stage',
* 'largest_extrapelvic_peritoneal_focus',
* 'last_known_disease_status',
* 'laterality',
* 'lymph_node_involved_site',
* 'lymph_nodes_positive',
* 'lymph_nodes_tested',
* 'lymphatic_invasion_present',
* 'margin_distance',
* 'margins_involved_site',
* 'masaoka_stage',
* 'medulloblastoma_molecular_classification',
* 'metastasis_at_diagnosis',
* 'metastasis_at_diagnosis_site',
* 'method_of_diagnosis',
* 'micropapillary_features',
* 'mitosis_karyorrhexis_index',
* 'mitotic_count',
* 'morphology',
* 'non_nodal_regional_disease',
* 'non_nodal_tumor_deposits',
* 'ovarian_specimen_status',
* 'ovarian_surface_involvement',
* 'papillary_renal_cell_type',
* 'percent_tumor_invasion',
* 'perineural_invasion_present',
* 'peripancreatic_lymph_nodes_positive',
* 'peripancreatic_lymph_nodes_tested',
* 'peritoneal_fluid_cytological_status',
* 'pregnant_at_diagnosis',
* 'primary_diagnosis',
* 'primary_disease',
* 'primary_gleason_grade',
* 'prior_malignancy',
* 'prior_treatment',
* 'progression_or_recurrence',
* 'residual_disease',
* 'satellite_nodule_present',
* 'secondary_gleason_grade',
* 'site_of_resection_or_biopsy',
* 'sites_of_involvement',
* 'supratentorial_localization',
* 'synchronous_malignancy',
* 'tissue_or_organ_of_origin',
* 'transglottic_ex*tension',
* 'tumor_confined_to_organ_of_origin',
* 'tumor_depth',
* 'tumor_focality',
* 'tumor_grade',
* 'tumor_largest_dimension_diameter',
* 'tumor_regression_grade',
* 'tumor_stage',
* 'vascular_invasion_present',
* 'vascular_invasion_type',
* 'weiss_assessment_score',
* 'who_cns_grade',
* 'who_nte_grade',
* 'wilms_tumor_histologic_subtype',
* 'year_of_diagnosis',
* 'chemo_concurrent_to_radiation',
* 'days_to_treatment_end',
* 'days_to_treatment_start',
* 'initial_disease_status',
* 'number_of_cycles',
* 'reason_treatment_ended',
* 'regimen_or_line_of_therapy',
* 'route_of_administration',
* 'therapeutic_agents',
* 'treatment_anatomic_site',
* 'treatment_arm',
* 'treatment_dose',
* 'treatment_dose_units',
* 'treatment_effect',
* 'treatment_effect_indicator',
* 'treatment_frequency',
* 'treatment_intent_type',
* 'treatment_or_therapy',
* 'treatment_outcome',
* 'treatment_type'

In [4]:
clin = pd.read_csv('/content/clinical.tsv', sep=None, engine='python')
# clin[clin == "'--"] = None
# clin.dropna(axis= 1, how= 'all', inplace= True)
# clin.describe()

In [5]:
# clin = pd.read_csv('/content/clinical.tsv', sep=None, engine='python')
# clin

In [6]:
# clin[clin == "'--"] = None

In [7]:
# for name, values in clin.items():
#   if clin[name].all() == None:
#     clin.dropna([name], axis= 1, how= 'all', inplace= True)

In [8]:
# clin.dropna(axis= 1, how= 'all', inplace= True)

In [9]:
# for name, values in clin.items():
#   if clin[name].any == False:
#     clin.drop([name], axis= 1, how= 'all', inplace= True)


In [10]:
# clin['cause_of_death'].unique()

In [11]:
# for n in clin.columns:
#   if clin[n].any() == False:
#     clin.dropna(axis='columns',how='all', inplace=True)

In [12]:
clin.shape

(106, 158)

In [13]:
# clin.columns.values.tolist ()

In [14]:
clin.describe()

case_id case_submitter_id project_id  \
count                                    106               106        106   
unique                                    58                58          1   
top     0030a28c-81aa-44b0-8be0-b35e1dcbf98c      TCGA-FA-A7Q1  TCGA-DLBC   
freq                                       2                 2        106   

       age_at_index age_is_obfuscated cause_of_death cause_of_death_source  \
count           106               106            106                   106   
unique           35                 1              1                     1   
top             '--               '--            '--                   '--   
freq             10               106            106                   106   

       country_of_residence_at_enrollment days_to_birth days_to_death  ...  \
count                                 106           106           106  ...   
unique                                  1            49            10  ...   
top                                   '--           '--           '--  ...   
freq                                  106            10            88  ...   

       treatment_arm treatment_dose treatment_dose_units treatment_effect  \
count            106            106                  106              106   
unique             1              1                    1                1   
top              '--            '--                  '--              '--   
freq             106            106                  106              106   

       treatment_effect_indicator treatment_frequency treatment_intent_type  \
count                         106                 106                   106   
unique                          1                   1                     1   
top                           '--                 '--                   '--   
freq                          106                 106                   106   

       treatment_or_therapy treatment_outcome               treatment_type  
count                   106               106                          106  
unique                    4                 1                            3  
top                     yes               '--  Pharmaceutical Therapy, NOS  
freq                     51               106                           48  

[4 rows x 158 columns]

In [15]:
# clin.loc[clin['ajcc_staging_system_edition'] == "'--", 'ajcc_staging_system_edition'] = ''

In [16]:
# clin['ajcc_staging_system_edition']

In [17]:
replacement = {"'--" : ""}
for name, values in clin.items ():
  values.map(replacement)



# print(n)

In [18]:

clin.loc[clin['ajcc_staging_system_edition'] != "'--"]

Empty DataFrame
Columns: [case_id, case_submitter_id, project_id, age_at_index, age_is_obfuscated, cause_of_death, cause_of_death_source, country_of_residence_at_enrollment, days_to_birth, days_to_death, ethnicity, gender, occupation_duration_years, premature_at_birth, race, vital_status, weeks_gestation_at_birth, year_of_birth, year_of_death, adrenal_hormone, age_at_diagnosis, ajcc_clinical_m, ajcc_clinical_n, ajcc_clinical_stage, ajcc_clinical_t, ajcc_pathologic_m, ajcc_pathologic_n, ajcc_pathologic_stage, ajcc_pathologic_t, ajcc_staging_system_edition, anaplasia_present, anaplasia_present_type, ann_arbor_b_symptoms, ann_arbor_b_symptoms_described, ann_arbor_clinical_stage, ann_arbor_extranodal_involvement, ann_arbor_pathologic_stage, best_overall_response, breslow_thickness, burkitt_lymphoma_clinical_variant, child_pugh_classification, circumferential_resection_margin, classification_of_tumor, cog_liver_stage, cog_neuroblastoma_risk_group, cog_renal_stage, cog_rhabdomyosarcoma_risk_group, days_to_best_overall_response, days_to_diagnosis, days_to_last_follow_up, days_to_last_known_disease_status, days_to_recurrence, eln_risk_classification, enneking_msts_grade, enneking_msts_metastasis, enneking_msts_stage, enneking_msts_tumor_site, esophageal_columnar_dysplasia_degree, esophageal_columnar_metaplasia_present, figo_stage, figo_staging_edition_year, first_symptom_prior_to_diagnosis, gastric_esophageal_junction_involvement, gleason_grade_group, gleason_grade_tertiary, gleason_patterns_percent, goblet_cells_columnar_mucosa_present, greatest_tumor_dimension, gross_tumor_weight, icd_10_code, igcccg_stage, inpc_grade, inpc_histologic_group, inrg_stage, inss_stage, international_prognostic_index, irs_group, irs_stage, ishak_fibrosis_score, iss_stage, largest_extrapelvic_peritoneal_focus, last_known_disease_status, laterality, lymph_node_involved_site, lymph_nodes_positive, lymph_nodes_tested, lymphatic_invasion_present, margin_distance, margins_involved_site, masaoka_stage, medulloblastoma_molecular_classification, metastasis_at_diagnosis, metastasis_at_diagnosis_site, method_of_diagnosis, micropapillary_features, mitosis_karyorrhexis_index, mitotic_count, morphology, non_nodal_regional_disease, non_nodal_tumor_deposits, ...]
Index: []

[0 rows x 158 columns]

##Видимо Группы риска (вредные привычки).
#!NB Самая старая женьщина в Мире всю жизнь курила и лопала шоколад (прожила 120 лет)

Колонки:
* 'case_id' ,
* 'case_submitter_id',
* 'project_id',
* 'age_at_onset',
* 'alcohol_days_per_week',
* 'alcohol_drinks_per_day',
* 'alcohol_history',
* 'alcohol_intensity',
* 'alcohol_type',
* 'asbestos_exposure',
* 'bmi',
* 'cigarettes_per_day',
* 'coal_dust_exposure',
* 'environmental_tobacco_smoke_exposure',
* 'exposure_duration',
* 'exposure_duration_years',
* 'exposure_type',
* 'height',
* 'marijuana_use_per_week',
* 'pack_years_smoked',
* 'parent_with_radiation_exposure',
* 'radon_exposure',
* 'respirable_crystalline_silica_exposure',
* 'secondhand_smoke_as_child',
* 'smokeless_tobacco_quit_age',
* 'smoking_frequency',
* 'time_between_waking_and_first_smoke',
* 'tobacco_smoking_onset_year',
* 'tobacco_smoking_quit_year',
* 'tobacco_smoking_status',
* 'tobacco_use_per_day',
* 'type_of_smoke_exposure',
* 'type_of_tobacco_used',
* 'weight',
* 'years_smoked'

In [19]:
exposure = pd.read_csv('/content/exposure.tsv', sep=None)
# exposure

<ipython-input-19-91fbf7ee888d>:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support sep=None with delim_whitespace=False; you can avoid this warning by specifying engine='python'.
  exposure = pd.read_csv('/content/exposure.tsv', sep=None)


In [20]:
# exposure.columns.values.tolist ()

In [21]:
exposure.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48 entries, 0 to 47
Data columns (total 35 columns):
 #   Column                                  Non-Null Count  Dtype 
---  ------                                  --------------  ----- 
 0   case_id                                 48 non-null     object
 1   case_submitter_id                       48 non-null     object
 2   project_id                              48 non-null     object
 3   age_at_onset                            48 non-null     object
 4   alcohol_days_per_week                   48 non-null     object
 5   alcohol_drinks_per_day                  48 non-null     object
 6   alcohol_history                         48 non-null     object
 7   alcohol_intensity                       48 non-null     object
 8   alcohol_type                            48 non-null     object
 9   asbestos_exposure                       48 non-null     object
 10  bmi                                     48 non-null     object
 11  cigarett

In [22]:
exposure[exposure == "'--"] = None
exposure.dropna(axis= 1, how= 'all', inplace= True)
exposure.info()
# clin.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48 entries, 0 to 47
Data columns (total 4 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   case_id            48 non-null     object
 1   case_submitter_id  48 non-null     object
 2   project_id         48 non-null     object
 3   alcohol_history    48 non-null     object
dtypes: object(4)
memory usage: 1.6+ KB


##Видимо Семейный анамнез (история семьи)
Но записей нет
Столбцы:
* 'case_id',
* 'case_submitter_id',
* 'project_id',
* 'relationship_age_at_diagnosis',
* 'relationship_gender',
* 'relationship_primary_diagnosis',
* 'relationship_type',
* 'relative_with_cancer_history',
* 'relatives_with_cancer_history_count'

In [23]:
fam_hist = pd.read_csv('/content/family_history.tsv', sep= None)
# fam_hist.info()

<ipython-input-23-b54e295bc04e>:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support sep=None with delim_whitespace=False; you can avoid this warning by specifying engine='python'.
  fam_hist = pd.read_csv('/content/family_history.tsv', sep= None)


##Видимо Исход (дальнейшее наблюдение, диспансеризация)
Но записей нет

Столбцы:
* 'case_id',
* 'case_submitter_id',
* 'project_id',
* 'adverse_event',
* 'adverse_event_grade',
* 'aids_risk_factors',
* 'barretts_esophagus_goblet_cells_present',
* 'bmi',
* 'body_surface_area',
* 'cause_of_response',
* 'cd4_count',
* 'cdc_hiv_risk_factors',
* 'comorbidity',
* 'comorbidity_method_of_diagnosis',
* 'days_to_adverse_event',
* 'days_to_comorbidity',
* 'days_to_follow_up',
* 'days_to_imaging',
* 'days_to_progression',
* 'days_to_progression_free',
* 'days_to_recurrence',
* 'diabetes_treatment_type',
* 'disease_response',
* 'dlco_ref_predictive_percent',
* 'ecog_performance_status',
* 'evidence_of_recurrence_type',
* 'eye_color',
* 'fev1_fvc_post_bronch_percent',
* 'fev1_fvc_pre_bronch_percent',
* 'fev1_ref_post_bronch_percent',
* 'fev1_ref_pre_bronch_percent',
* 'follow_up_id',
* 'follow_up_submitter_id',
* 'haart_treatment_indicator',
* 'height',
* 'hepatitis_sustained_virological_response',
* 'history_of_tumor',
* 'history_of_tumor_type',
* 'hiv_viral_load',
* 'hormonal_contraceptive_type',
* 'hormonal_contraceptive_use',
* 'hormone_replacement_therapy_type',
* 'hpv_positive_type',
* 'hysterectomy_margins_involved',
* 'hysterectomy_type',
* 'imaging_result',
* 'imaging_type',
* 'immunosuppressive_treatment_type',
* 'karnofsky_performance_status',
* 'menopause_status',
* 'nadir_cd4_count',
* 'pancreatitis_onset_year',
* 'pregnancy_outcome',
* 'procedures_performed',
* 'progression_or_recurrence',
* 'progression_or_recurrence_anatomic_site',
* 'progression_or_recurrence_type',
* 'recist_targeted_regions_number',
* 'recist_targeted_regions_sum',
* 'reflux_treatment_type',
* 'risk_factor',
* 'risk_factor_treatment',
* 'scan_tracer_used',
* 'undescended_testis_corrected',
* 'undescended_testis_corrected_age',
* 'undescended_testis_corrected_laterality',
* 'undescended_testis_corrected_method',
* 'undescended_testis_history',
* 'undescended_testis_history_laterality',
* 'viral_hepatitis_serologies',
* 'weight',
* 'aa_change',
* 'antigen',
* 'biospecimen_type',
* 'biospecimen_volume',
* 'blood_test_normal_range_lower',
* 'blood_test_normal_range_upper',
* 'cell_count',
* 'chromosome',
* 'clonality',
* 'copy_number',
* 'cytoband',
* 'days_to_test',
* 'exon',
* 'gene_symbol',
* 'histone_family',
* 'histone_variant',
* 'intron',
* 'laboratory_test',
* 'loci_abnormal_count',
* 'loci_count',
* 'locus',
* 'mismatch_repair_mutation',
* 'mitotic_count',
* 'mitotic_total_area',
* 'molecular_analysis_method',
* 'molecular_consequence',
* 'molecular_test_id',
* 'molecular_test_submitter_id',
* 'pathogenicity',
* 'ploidy',
* 'second_exon',
* 'second_gene_symbol',
* 'specialized_molecular_test',
* 'test_analyte_type',
* 'test_result',
* 'test_units',
* 'test_value',
* 'transcript',
* 'variant_origin',
* 'variant_type',
* 'zygosity'


In [24]:
foll_up = pd.read_csv('/content/follow_up.tsv', sep= None)
# foll_up

<ipython-input-24-f5634306870d>:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support sep=None with delim_whitespace=False; you can avoid this warning by specifying engine='python'.
  foll_up = pd.read_csv('/content/follow_up.tsv', sep= None)


In [25]:
# foll_up.columns.values.tolist()

In [26]:
# foll_up.info()

##Видимо Подробности патологии
Но тоже пустая

Столбцы:
* 'case_id',
* 'case_submitter_id',
* 'project_id',
* 'diagnosis_id',
* 'diagnosis_submitter_id',
* 'additional_pathology_findings',
* 'anaplasia_present',
* 'anaplasia_present_type',
* 'bone_marrow_malignant_cells',
* 'breslow_thickness',
* 'circumferential_resection_margin',
* 'columnar_mucosa_present',
* 'consistent_pathology_review',
* 'dysplasia_degree',
* 'dysplasia_type',
* 'greatest_tumor_dimension',
* 'gross_tumor_weight',
* 'largest_extrapelvic_peritoneal_focus',
* 'lymph_node_involved_site',
* 'lymph_node_involvement',
* 'lymph_nodes_positive',
* 'lymph_nodes_tested',
* 'lymphatic_invasion_present',
* 'margin_status',
* 'metaplasia_present',
* 'morphologic_architectural_pattern',
* 'necrosis_percent',
* 'necrosis_present',
* 'non_nodal_regional_disease',
* 'non_nodal_tumor_deposits',
* 'number_proliferating_cells',
* 'pathology_detail_id',
* 'pathology_detail_submitter_id',
* 'percent_tumor_invasion',
* 'perineural_invasion_present',
* 'peripancreatic_lymph_nodes_positive',
* 'peripancreatic_lymph_nodes_tested',
* 'prostatic_chips_positive_count',
* 'prostatic_chips_total_count',
* 'prostatic_involvement_percent',
* 'residual_tumor',
* 'rhabdoid_percent',
* 'rhabdoid_present',
* 'sarcomatoid_percent',
* 'sarcomatoid_present',
* 'size_extraocular_nodule',
* 'transglottic_extension',
* 'tumor_largest_dimension_diameter',
* 'tumor_thickness',
* 'vascular_invasion_present',
* 'vascular_invasion_type'

In [27]:
# path_det = pd.read_csv('/content/pathology_detail.tsv', sep=None)
# path_det

In [28]:
# path_det.columns.values.tolist()

In [29]:
# path_det.info()

#Заключение
В архиве представлено 5 таблиц (Клиника, Группы риска, Семейный анамнез, Исход, Подробности патологии), 3 таблицы не заполнены. Поэтому в работу их не беру. Итого в работе 2 таблицы (Клиника(***clinical.tsv***) и Группы риска(***exposure.tsv***)).
Для началы выберу столбцы в каждой таблице и объединю их по ***case_id***

In [30]:
clin.describe()

case_id case_submitter_id project_id  \
count                                    106               106        106   
unique                                    58                58          1   
top     0030a28c-81aa-44b0-8be0-b35e1dcbf98c      TCGA-FA-A7Q1  TCGA-DLBC   
freq                                       2                 2        106   

       age_at_index age_is_obfuscated cause_of_death cause_of_death_source  \
count           106               106            106                   106   
unique           35                 1              1                     1   
top             '--               '--            '--                   '--   
freq             10               106            106                   106   

       country_of_residence_at_enrollment days_to_birth days_to_death  ...  \
count                                 106           106           106  ...   
unique                                  1            49            10  ...   
top                                   '--           '--           '--  ...   
freq                                  106            10            88  ...   

       treatment_arm treatment_dose treatment_dose_units treatment_effect  \
count            106            106                  106              106   
unique             1              1                    1                1   
top              '--            '--                  '--              '--   
freq             106            106                  106              106   

       treatment_effect_indicator treatment_frequency treatment_intent_type  \
count                         106                 106                   106   
unique                          1                   1                     1   
top                           '--                 '--                   '--   
freq                          106                 106                   106   

       treatment_or_therapy treatment_outcome               treatment_type  
count                   106               106                          106  
unique                    4                 1                            3  
top                     yes               '--  Pharmaceutical Therapy, NOS  
freq                     51               106                           48  

[4 rows x 158 columns]

###Начнем удалять столбцы с одним уникальным значением

In [31]:
clin[clin == "'--"] = None
clin.dropna(axis= 1, how= 'all', inplace= True)
clin.describe()

case_id case_submitter_id project_id  \
count                                    106               106        106   
unique                                    58                58          1   
top     0030a28c-81aa-44b0-8be0-b35e1dcbf98c      TCGA-FA-A7Q1  TCGA-DLBC   
freq                                       2                 2        106   

       age_at_index days_to_birth days_to_death               ethnicity  \
count            96            96            18                      96   
unique           34            48             9                       2   
top              67        -22454           248  not hispanic or latino   
freq              6             2             2                      72   

        gender   race vital_status  ... prior_malignancy prior_treatment  \
count       96     96           96  ...               96              96   
unique       2      3            2  ...                1               2   
top     female  white        Alive  ...               no              No   
freq        52     58           78  ...               96              94   

       progression_or_recurrence         site_of_resection_or_biopsy  \
count                         96                                  96   
unique                         1                                  20   
top                 not reported  Lymph nodes of head, face and neck   
freq                          96                                  32   

       synchronous_malignancy           tissue_or_organ_of_origin  \
count                      96                                  96   
unique                      1                                  20   
top                        No  Lymph nodes of head, face and neck   
freq                       96                                  32   

         tumor_grade year_of_diagnosis treatment_or_therapy  \
count             96                96                   96   
unique             1                14                    3   
top     not reported              2013                  yes   
freq              96                18                   51   

                     treatment_type  
count                            96  
unique                            2  
top     Pharmaceutical Therapy, NOS  
freq                             48  

[4 rows x 33 columns]

In [32]:
clin.columns.values.tolist()

['case_id',
 'case_submitter_id',
 'project_id',
 'age_at_index',
 'days_to_birth',
 'days_to_death',
 'ethnicity',
 'gender',
 'race',
 'vital_status',
 'year_of_birth',
 'year_of_death',
 'age_at_diagnosis',
 'ann_arbor_b_symptoms',
 'ann_arbor_clinical_stage',
 'ann_arbor_extranodal_involvement',
 'classification_of_tumor',
 'days_to_diagnosis',
 'days_to_last_follow_up',
 'icd_10_code',
 'last_known_disease_status',
 'morphology',
 'primary_diagnosis',
 'prior_malignancy',
 'prior_treatment',
 'progression_or_recurrence',
 'site_of_resection_or_biopsy',
 'synchronous_malignancy',
 'tissue_or_organ_of_origin',
 'tumor_grade',
 'year_of_diagnosis',
 'treatment_or_therapy',
 'treatment_type']

Остались столбцы:
* 'case_id'- ***идентификатор амбулаторной/стационарной карты***,
* 'case_submitter_id' - ***идентификатор источника данных***,
* 'project_id' - ***идентификатор проекта***,
* 'age_at_index' - ***возраст на момент описания клиники***,
* 'days_to_birth' - ***дата рождения***,
* 'days_to_death' - ***дата смерти***,
* 'ethnicity' - ***этническая принадлежность***,
* 'gender' - ***пол***,
* 'race' - ***расса***,
* 'vital_status' - ***жизненый уровень***,
* 'year_of_birth' - ***год рождения***,
* 'year_of_death' - ***год смерти***,
* 'age_at_diagnosis' - ***продолжительность болезни***,
* 'ann_arbor_b_symptoms' - ***стадирование симптомов по системе ANN-Arbor***,
* 'ann_arbor_clinical_stage' - ***стадирование клиники по системе ANN-Arbor***,
* 'ann_arbor_extranodal_involvement'- ***стадирование экстранодальное поражение по системе ANN-Arbor***,
* 'classification_of_tumor' - ***классификация опухолей***,
* 'days_to_diagnosis' - ***дней до диагностики***,
* 'days_to_last_follow_up' - ***дней до последнего наблюдения***,
* 'icd_10_code' - ***код по МКБ-10***,
* 'last_known_disease_status' - ***последний известный статус заболевания***,
* 'morphology' - ***морфология***,
* 'primary_diagnosis' - ***первичный диагноз***,
* 'prior_malignancy' - ***предшествующее злокачественное новообразование***,
* 'prior_treatment' - ***предшествующее лечение***,
* 'progression_or_recurrence' - ***прогрессирование или рецидив***,
* 'site_of_resection_or_biopsy' - ***место резекции или биопсии***,
* 'synchronous_malignancy' - ***дополнительное злокачественное новообразование***,
* 'tissue_or_organ_of_origin' - ***ткань или орган происхождения***,
* 'tumor_grade' - ***степень опухоли***,
* 'year_of_diagnosis' - ***год постановки диагноза***,
* 'treatment_or_therapy' - ***консервативное или оперативное лечение***,
* 'treatment_type' - ***тип терапии***
их и переведу.

In [33]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
pd.set_option('display.max_colwidth', None)

In [34]:
print(clin['age_at_index'].isna)

<bound method Series.isna of 0        61
1        61
2        73
3        73
4        27
5        27
6        60
7        60
8        67
9        67
10       72
11       72
12       45
13       45
14     None
15       28
16       28
17     None
18       63
19       63
20       23
21       23
22       46
23       46
24       57
25       57
26       70
27       70
28       67
29       67
30       58
31       58
32     None
33     None
34       69
35       69
36     None
37       63
38       63
39       37
40       37
41       66
42       66
43       51
44       51
45       58
46       58
47       43
48       43
49       82
50       82
51       56
52       56
53       75
54       75
55       67
56       67
57       40
58       40
59       55
60       55
61       70
62       70
63       64
64       64
65       54
66       54
67       36
68       36
69     None
70       40
71       40
72       42
73       42
74     None
75     None
76       57
77       57
78     None
79       62
80       62

In [35]:
clin.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 106 entries, 0 to 105
Data columns (total 33 columns):
 #   Column                            Non-Null Count  Dtype 
---  ------                            --------------  ----- 
 0   case_id                           106 non-null    object
 1   case_submitter_id                 106 non-null    object
 2   project_id                        106 non-null    object
 3   age_at_index                      96 non-null     object
 4   days_to_birth                     96 non-null     object
 5   days_to_death                     18 non-null     object
 6   ethnicity                         96 non-null     object
 7   gender                            96 non-null     object
 8   race                              96 non-null     object
 9   vital_status                      96 non-null     object
 10  year_of_birth                     96 non-null     object
 11  year_of_death                     10 non-null     object
 12  age_at_diagnosis      

В данной таблице 106 записей, только у 96 пациентов указан возраст и код по МКБ-10. Похоже в работу нужно взять только этих пациентов.

In [36]:
# df.dropna (subset=['assists'])
# df = df[df['ATR2'].notnull()]
from pandas.core.window.ewm import isna
clin_w = clin.dropna(subset=['age_at_index'])
clin_w.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 96 entries, 0 to 105
Data columns (total 33 columns):
 #   Column                            Non-Null Count  Dtype 
---  ------                            --------------  ----- 
 0   case_id                           96 non-null     object
 1   case_submitter_id                 96 non-null     object
 2   project_id                        96 non-null     object
 3   age_at_index                      96 non-null     object
 4   days_to_birth                     96 non-null     object
 5   days_to_death                     18 non-null     object
 6   ethnicity                         96 non-null     object
 7   gender                            96 non-null     object
 8   race                              96 non-null     object
 9   vital_status                      96 non-null     object
 10  year_of_birth                     96 non-null     object
 11  year_of_death                     10 non-null     object
 12  age_at_diagnosis       

In [37]:
clin_w.head(10)

case_id case_submitter_id project_id  \
0  0030a28c-81aa-44b0-8be0-b35e1dcbf98c      TCGA-FA-A7Q1  TCGA-DLBC   
1  0030a28c-81aa-44b0-8be0-b35e1dcbf98c      TCGA-FA-A7Q1  TCGA-DLBC   
2  0bf573ac-cd1e-42d8-90cf-b30d7b08679c      TCGA-FA-A6HN  TCGA-DLBC   
3  0bf573ac-cd1e-42d8-90cf-b30d7b08679c      TCGA-FA-A6HN  TCGA-DLBC   
4  0e251c03-bf86-4ed8-b45d-3cbc97160502      TCGA-GS-A9U4  TCGA-DLBC   
5  0e251c03-bf86-4ed8-b45d-3cbc97160502      TCGA-GS-A9U4  TCGA-DLBC   
6  0e9fcccc-0630-408d-a121-2c6413824cb7      TCGA-FF-8062  TCGA-DLBC   
7  0e9fcccc-0630-408d-a121-2c6413824cb7      TCGA-FF-8062  TCGA-DLBC   
8  1843c82e-7a35-474f-9f79-c0a9af9aa09c      TCGA-FA-A4BB  TCGA-DLBC   
9  1843c82e-7a35-474f-9f79-c0a9af9aa09c      TCGA-FA-A4BB  TCGA-DLBC   

  age_at_index days_to_birth days_to_death               ethnicity  gender  \
0           61        -22454           248  not hispanic or latino  female   
1           61        -22454           248  not hispanic or latino  female   
2           73        -27025            58  not hispanic or latino    male   
3           73        -27025            58  not hispanic or latino    male   
4           27        -10075           391      hispanic or latino  female   
5           27        -10075           391      hispanic or latino  female   
6           60        -21959          None  not hispanic or latino  female   
7           60        -21959          None  not hispanic or latino  female   
8           67        -24590          None  not hispanic or latino    male   
9           67        -24590          None  not hispanic or latino    male   

    race vital_status year_of_birth year_of_death age_at_diagnosis  \
0  asian         Dead          1952          None            22454   
1  asian         Dead          1952          None            22454   
2  asian         Dead          1940          None            27025   
3  asian         Dead          1940          None            27025   
4  white         Dead          1984          2012            10075   
5  white         Dead          1984          2012            10075   
6  asian        Alive          1951          None            21959   
7  asian        Alive          1951          None            21959   
8  white        Alive          1945          None            24590   
9  white        Alive          1945          None            24590   

  ann_arbor_b_symptoms ann_arbor_clinical_stage  \
0                   No                  Stage I   
1                   No                  Stage I   
2                   No                     None   
3                   No                     None   
4                   No                 Stage II   
5                   No                 Stage II   
6                   No                  Stage I   
7                   No                  Stage I   
8                  Yes                 Stage IV   
9                  Yes                 Stage IV   

  ann_arbor_extranodal_involvement classification_of_tumor days_to_diagnosis  \
0                               No            not reported                 0   
1                               No            not reported                 0   
2                               No            not reported                 0   
3                               No            not reported                 0   
4                               No            not reported                 0   
5                               No            not reported                 0   
6                              Yes            not reported                 0   
7                              Yes            not reported                 0   
8                              Yes            not reported                 0   
9                              Yes            not reported                 0   

  days_to_last_follow_up icd_10_code last_known_disease_status morphology  \
0                     22       C48.1              not reported     9680/3   
1           

Получили подчищенную табличку, с которой уже можно работать.

Изменим типы объектов

In [38]:
for colname in ['age_at_diagnosis', 'days_to_birth', 'year_of_birth', 'age_at_diagnosis', 'days_to_diagnosis','year_of_diagnosis']:
  clin_w[colname] =  clin_w[colname].astype('int')

<ipython-input-38-b810250096ad>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  clin_w[colname] =  clin_w[colname].astype('int')


In [39]:
# clin_w['age_at_diagnosis', 'days_to_birth', 'days_to_death', 'year_of_birth', 'year_of_death', 'age_at_diagnosis',
      #  'days_to_diagnosis', 'days_to_last_follow_up', 'year_of_diagnosis'].astype('int')

In [40]:
clin_w['age_at_diagnosis'] = clin_w['age_at_diagnosis'].astype('int')
clin_w['days_to_birth'] = clin_w['days_to_birth'].astype('int')

<ipython-input-40-d620ff810fd7>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  clin_w['age_at_diagnosis'] = clin_w['age_at_diagnosis'].astype('int')
<ipython-input-40-d620ff810fd7>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  clin_w['days_to_birth'] = clin_w['days_to_birth'].astype('int')


In [41]:
clin_w.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 96 entries, 0 to 105
Data columns (total 33 columns):
 #   Column                            Non-Null Count  Dtype 
---  ------                            --------------  ----- 
 0   case_id                           96 non-null     object
 1   case_submitter_id                 96 non-null     object
 2   project_id                        96 non-null     object
 3   age_at_index                      96 non-null     object
 4   days_to_birth                     96 non-null     int64 
 5   days_to_death                     18 non-null     object
 6   ethnicity                         96 non-null     object
 7   gender                            96 non-null     object
 8   race                              96 non-null     object
 9   vital_status                      96 non-null     object
 10  year_of_birth                     96 non-null     int64 
 11  year_of_death                     10 non-null     object
 12  age_at_diagnosis       

In [42]:
clin_w.describe()

days_to_birth  year_of_birth  age_at_diagnosis  days_to_diagnosis  \
count      96.000000      96.000000         96.000000               96.0   
mean   -20731.354167    1952.875000      20731.354167                0.0   
std      5080.518956      15.092173       5080.518956                0.0   
min    -30256.000000    1923.000000       8493.000000                0.0   
25%    -24607.750000    1941.000000      16998.250000                0.0   
50%    -21146.500000    1951.500000      21146.500000                0.0   
75%    -16998.250000    1963.250000      24607.750000                0.0   
max     -8493.000000    1987.000000      30256.000000                0.0   

       year_of_diagnosis  
count          96.000000  
mean         2009.145833  
std             5.005217  
min          1987.000000  
25%          2009.000000  
50%          2011.000000  
75%          2012.000000  
max          2013.000000

In [43]:
clin_w.describe(include= 'all')

case_id case_submitter_id project_id  \
count                                     96                96         96   
unique                                    48                48          1   
top     0030a28c-81aa-44b0-8be0-b35e1dcbf98c      TCGA-FA-A7Q1  TCGA-DLBC   
freq                                       2                 2         96   
mean                                     NaN               NaN        NaN   
std                                      NaN               NaN        NaN   
min                                      NaN               NaN        NaN   
25%                                      NaN               NaN        NaN   
50%                                      NaN               NaN        NaN   
75%                                      NaN               NaN        NaN   
max                                      NaN               NaN        NaN   

       age_at_index  days_to_birth days_to_death               ethnicity  \
count            96      96.000000            18                      96   
unique           34            NaN             9                       2   
top              67            NaN           248  not hispanic or latino   
freq              6            NaN             2                      72   
mean            NaN  -20731.354167           NaN                     NaN   
std             NaN    5080.518956           NaN                     NaN   
min             NaN  -30256.000000           NaN                     NaN   
25%             NaN  -24607.750000           NaN                     NaN   
50%             NaN  -21146.500000           NaN                     NaN   
75%             NaN  -16998.250000           NaN                     NaN   
max             NaN   -8493.000000           NaN                     NaN   

        gender   race vital_status  year_of_birth year_of_death  \
count       96     96           96      96.000000            10   
unique       2      3            2            NaN             4   
top     female  white        Alive            NaN          2012   
freq        52     58           78            NaN             4   
mean       NaN    NaN          NaN    1952.875000           NaN   
std        NaN    NaN          NaN      15.092173           NaN   
min        NaN    NaN          NaN    1923.000000           NaN   
25%        NaN    NaN          NaN    1941.000000           NaN   
50%        NaN    NaN          NaN    1951.500000           NaN   
75%        NaN    NaN          NaN    1963.250000           NaN   
max        NaN    NaN          NaN    1987.000000           NaN   

        age_at_diagnosis ann_arbor_b_symptoms ann_arbor_clinical_stage  \
count          96.000000                   92                       84   
unique               NaN                    2                        4   
top                  NaN                   No                 Stage II   
freq                 NaN                   70                       34   
mean        20731.354167                  NaN                      NaN   
std          5080.518956                  NaN                      NaN   
min          8493.000000                  NaN                      NaN   
25%         16998.250000                  NaN                      NaN   
50%         21146.500000                  NaN                      NaN   
75%         24607.750000                  NaN                      NaN   
max         30256.000000                  NaN                      NaN   

       ann_arbor_extranodal_involvement classification_of_tumor  \
count                                92                      96   
unique                                2                       1   
top                                  No            not reported   
freq                                 50                      96   
mean                                NaN                     NaN   
std                                 NaN                     NaN   
min                                 

In [44]:
clin_w.describe(percentiles= [0.05, 0.25, 0.5, 0.75, 0.95])

days_to_birth  year_of_birth  age_at_diagnosis  days_to_diagnosis  \
count      96.000000      96.000000         96.000000               96.0   
mean   -20731.354167    1952.875000      20731.354167                0.0   
std      5080.518956      15.092173       5080.518956                0.0   
min    -30256.000000    1923.000000       8493.000000                0.0   
5%     -27468.000000    1929.000000      10466.000000                0.0   
25%    -24607.750000    1941.000000      16998.250000                0.0   
50%    -21146.500000    1951.500000      21146.500000                0.0   
75%    -16998.250000    1963.250000      24607.750000                0.0   
95%    -10466.000000    1982.000000      27468.000000                0.0   
max     -8493.000000    1987.000000      30256.000000                0.0   

       year_of_diagnosis  
count          96.000000  
mean         2009.145833  
std             5.005217  
min          1987.000000  
5%           2000.000000  
25%          2009.000000  
50%          2011.000000  
75%          2012.000000  
95%          2013.000000  
max          2013.000000

In [45]:
clin_w = clin_w.drop_duplicates()
clin_w

case_id case_submitter_id project_id  \
0    0030a28c-81aa-44b0-8be0-b35e1dcbf98c      TCGA-FA-A7Q1  TCGA-DLBC   
1    0030a28c-81aa-44b0-8be0-b35e1dcbf98c      TCGA-FA-A7Q1  TCGA-DLBC   
2    0bf573ac-cd1e-42d8-90cf-b30d7b08679c      TCGA-FA-A6HN  TCGA-DLBC   
3    0bf573ac-cd1e-42d8-90cf-b30d7b08679c      TCGA-FA-A6HN  TCGA-DLBC   
4    0e251c03-bf86-4ed8-b45d-3cbc97160502      TCGA-GS-A9U4  TCGA-DLBC   
5    0e251c03-bf86-4ed8-b45d-3cbc97160502      TCGA-GS-A9U4  TCGA-DLBC   
6    0e9fcccc-0630-408d-a121-2c6413824cb7      TCGA-FF-8062  TCGA-DLBC   
7    0e9fcccc-0630-408d-a121-2c6413824cb7      TCGA-FF-8062  TCGA-DLBC   
8    1843c82e-7a35-474f-9f79-c0a9af9aa09c      TCGA-FA-A4BB  TCGA-DLBC   
9    1843c82e-7a35-474f-9f79-c0a9af9aa09c      TCGA-FA-A4BB  TCGA-DLBC   
10   1d0db5d7-39ca-466d-96b3-0d278c5ea768      TCGA-FF-A7CW  TCGA-DLBC   
11   1d0db5d7-39ca-466d-96b3-0d278c5ea768      TCGA-FF-A7CW  TCGA-DLBC   
12   1ea575f1-f731-408b-a629-f5f4abab569e      TCGA-GS-A9TU  TCGA-DLBC   
13   1ea575f1-f731-408b-a629-f5f4abab569e      TCGA-GS-A9TU  TCGA-DLBC   
15   1f971af1-6772-4fe6-8d35-bbe527a037fe      TCGA-FF-8041  TCGA-DLBC   
16   1f971af1-6772-4fe6-8d35-bbe527a037fe      TCGA-FF-8041  TCGA-DLBC   
18   25ff86af-beb4-480c-b706-f3fe0306f7cf      TCGA-RQ-A68N  TCGA-DLBC   
19   25ff86af-beb4-480c-b706-f3fe0306f7cf      TCGA-RQ-A68N  TCGA-DLBC   
20   29aff186-c321-4ff9-b81b-105e27e620ff      TCGA-GS-A9TW  TCGA-DLBC   
21   29aff186-c321-4ff9-b81b-105e27e620ff      TCGA-GS-A9TW  TCGA-DLBC   
22   29e3d122-15a1-4235-a356-b1a9f94ceb39      TCGA-FF-A7CX  TCGA-DLBC   
23   29e3d122-15a1-4235-a356-b1a9f94ceb39      TCGA-FF-A7CX  TCGA-DLBC   
24   31bbad4e-3789-42ec-9faa-1cb86970f723      TCGA-G8-6907  TCGA-DLBC   
25   31bbad4e-3789-42ec-9faa-1cb86970f723      TCGA-G8-6907  TCGA-DLBC   
26   33365d22-cb83-4d8e-a2d1-06b675f75f6e      TCGA-GS-A9TT  TCGA-DLBC   
27   33365d22-cb83-4d8e-a2d1-06b675f75f6e      TCGA-GS-A9TT  TCGA-DLBC   
28   3622cf29-600f-4410-84d4-a9afeb41c475      TCGA-G8-6914  TCGA-DLBC   
29   3622cf29-600f-4410-84d4-a9afeb41c475      TCGA-G8-6914  TCGA-DLBC   
30   3f5a897d-1eaa-4d4c-8324-27ac07c90927      TCGA-G8-6906  TCGA-DLBC   
31   3f5a897d-1eaa-4d4c-8324-27ac07c90927      TCGA-G8-6906  TCGA-DLBC   
34   45b0cf9f-a879-417f-8f39-7770552252c0      TCGA-GS-A9TQ  TCGA-DLBC   
35   45b0cf9f-a879-417f-8f39-7770552252c0      TCGA-GS-A9TQ  TCGA-DLBC   
37   4dd86ebd-ef16-4b2b-9ea0-5d1d7afef257      TCGA-RQ-AAAT  TCGA-DLBC   
38   4dd86ebd-ef16-4b2b-9ea0-5d1d7afef257      TCGA-RQ-AAAT  TCGA-DLBC   
39   58e66976-4507-4552-ac53-83a49a142dde      TCGA-GS-A9TX  TCGA-DLBC   
40   58e66976-4507-4552-ac53-83a49a142dde      TCGA-GS-A9TX  TCGA-DLBC   
41   5eff68ff-f6c3-40c9-9fc8-00e684a7b712      TCGA-GR-A4D9  TCGA-DLBC   
42   5eff68ff-f6c3-40c9-9fc8-00e684a7b712      TCGA-GR-A4D9  TCGA-DLBC   
43   67325322-483f-443f-9ffa-2a20d108a2fb      TCGA-FF-8046  TCGA-DLBC   
44   67325322-483f-443f-9ffa-2a20d108a2fb      TCGA-FF-8046  TCGA-DLBC   
45   69f23725-adca-48ac-9b33-80a7aae24cfe      TCGA-FA-A6HO  TCGA-DLBC   
46   69f23725-adca-48ac-9b33-80a7aae24cfe      TCGA-FA-A6HO  TCGA-DLBC   
47   6a21c948-cd85-4150-8c01-83017d7dc1ed      TCGA-G8-6324  TCGA-DLBC   
48   6a21c948-cd85-4150-8c01-83017d7dc1ed      TCGA-G8-6324  TCGA-DLBC   
49   6e9437f0-a4ed-475c-ab0e-bf1431c70a90      TCGA-GS-A9TY  TCGA-DLBC   
50   6e9437f0-a4ed-475c-ab0e-bf1431c70a90      TCGA-GS-A9TY  TCGA-DLBC   
51   7a589441-11ef-4158-87e7-3951d86bc2aa      TCGA-GR-7351  TCGA-DLBC   
52   7a589441-11ef-4158-87e7-3951d86bc2aa      TCGA-GR-7351  TCGA-DLBC   
53   a43e5f0e-a21f-48d8-97e0-084d413680b7      TCGA-FA-8693  TCGA-DLBC   
54   a43e5f0e-a21f-48d8-97e0-084d413680b7      TCGA-FA-8693  TCGA-DLBC   
55   a468e725-ad4b-411d-ac5c-2eacc68ec580      TCGA-G8-6909  TCGA-DLBC   
56   a468e725-ad4b-411d-ac5c-2eacc68ec580      TCGA-G8-6909  TCGA-DLBC   
57   a5b188f0-a6d3-4d4a-b04f-36d47ec05338      TCGA-FA-A4XK  TCGA-DLBC   
58   a5b188f0-a6d3-4d4a-b04f-36d47ec05

Дублирования записей нет, просто пациенты получали два вида терапии: Радиотерапию и Химиотерапию. И это занесено в разные строки. Надо преобразовать и занести в одну строку. Для каждого пациента.

In [ ]:
clin_w['ter_01'] =